src/GameLoop.tsx

```diff
@@ -486,13 +478,19 @@ export function GameLoop() {
                                     cell.char
                                 ).join(""),
                     );
+                    totalMatchedWords++;
                     for (let i = col_top; i < col_bot + 1; ++i) {
                         matchedCells.add([i, c]);
                     }
                     hasRemovedWord = true;
                 }
             });
 
+            while (matchedWords.length > MAX_WORDLIST_LEN) {
+                matchedWords.shift();
+            }
```

Comment:
Very inefficient.  Use `Array.prototype.splice` or a queue.

Also, why not just display all the words?

Reply:
Removed